angular/projects/admin-nrpti/src/app/mines/mines-enforcement-actions/mines-court-convictions/mines-court-convictions-add-edit/mines-court-convictions-add-edit.component.ts

```diff
@@ -0,0 +1,170 @@
+      logger,
+      utils,
+      _changeDetectionRef,
+      storeService
+    );
+  }
+
+  ngOnInit() {
+    this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+      this.isEditing = res.breadcrumb !== 'Add Court Conviction';
+      if (this.isEditing) {
+        if (res && res.record && res.record[0] && res.record[0].data) {
+          this.currentRecord = res.record[0].data;
+          this.populateTextFields();
+        } else {
+          alert('Error: could not load edit mines cournt conviction.');
+          this.router.navigate([['mines', 'enforcement-actions']]);
+        }
+      } else {
+        this.currentRecord = {
+          sourceSystemRef: 'nrpti',
+          documents: [],
+          unlistedMine: ''
+        };
+      }
+
+      super.buildForm();
+      this.setRadioButtonSelection();
+      this.subscribeToFormControlChanges();
+      this.loading = false;
+      this._changeDetectionRef.detectChanges();
+    });
+  }
+
+  private setRadioButtonSelection() {
+    if (this.currentRecord && this.currentRecord.documents.length > 0) {
+      this.selectedConvictionInfoType = 'url';
+    } else if (this.currentRecord && this.bcmiFlavour && this.bcmiFlavour.description) {
+      this.selectedConvictionInfoType = 'text';
+    }
+
+    this.myForm.get('convictionInfoType').setValue(this.selectedConvictionInfoType);
+  }
+
+  protected subscribeToFormControlChanges() {
+    super.subscribeToFormControlChanges();
+
+    this.myForm
+      .get('convictionInfoType')
+      .valueChanges.pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(val => {
+        if (this.selectedConvictionInfoType && val !== this.selectedConvictionInfoType) {
+          const shouldCancel = confirm(
+            'Cancelling will discard unsaved changes. Are you sure you would like to continue?'
+          );
+          if (!shouldCancel) {
+            this.myForm.get('convictionInfoType').setValue(this.selectedConvictionInfoType);
+            // Hack to make radio button checked properly.  Without this hack the mouse click on the cancel button
+            // bugs out the checked radio button and make the UI display it incorrectly
+            setTimeout(() => {
+              this.myForm.get('convictionInfoType').setValue(this.selectedConvictionInfoType);
+            }, 1);
+          } else {
+            this.selectedConvictionInfoType = val;
+            this.myForm.get('convictionInfoType').setValue(val);
+
+            if (val === 'url') {
+              // Switching to url from text so reset staged prosecution text here
+              this.resetStagedProsecutionText();
+            } else {
+              // Switching from text to url so reset staged documents and links here
+              this.resetStagedLinks();
+            }
+          }
+        } else {
+          this.selectedConvictionInfoType = val;
+        }
+      });
+  }
+
+  private resetStagedLinks() {
+    this.documents = [];
```

Comment:
Editting a record with url link, if you switch to a text description the document link is not removed. Result is a record with both a link and description.
Is this intended?

Reply:
Yes this is intended.  The alert language says any `unsaved changes` will be discarded.  So the current behavior is to discard any staged changes on the document upload component.  Meaning any previously created links are untouched.